src/xml/node.cr

```diff
@@ -544,4 +552,20 @@ struct XML::Node
     ptr = @node.value._private
     ptr ? (ptr.as(Array(XML::Error))) : nil
   end
+
+  private SUBSTITUTIONS = {
+    '>'  => "&gt;",
+    '<'  => "&lt;",
+    '"'  => "&quot;",
+    '\'' => "&apos;",
+    '&'  => "&amp;",
+  }
+
+  private def escape(string)
+    if string.includes? '\0'
+      raise XML::Error.new("Cannot escape string containing NUL character", 0)
+    end
+
+    string.gsub(SUBSTITUTIONS)
```

Comment:
Why not use `HTML.escape`?

Reply:
Because it's not HTML?

Reply:
But it does the same thing, there is no need to implement this again. That's why `XML.escape` was removed in favour of `HTML.escape`.

Reply:
Okay, it's not *exactly* the same: `HTML.escape` encodes `'` as `&x39;` instead of `&apos;` because the named entity is not defined for HTML4. But it's still valid XML and no big deal. We could even consider changing the behaviour for `HTML.escape` because in other places (like `HTML.unescape`) we support only HTML5 anyway.

Reply:
I disagree, I really don't want to use a module called `HTML` in a module for `XML`. It's simply wrong. They're different standards and people working on the HTML escape method won't think about XML when they're modifying it. Besides, it makes very little difference either way because both methods are unlikely to change for a long long time.

Reply:
I see your point. But following this argumentation asks for reintroducing `XML.escape`.

Reply:
No, XML.escape was used because you shouldn't need it, because we have facilities to build XML built in. We don't have a HTML::Builder so HTML.escape has a purpose.

I wouldn't mind adding XML.escape back again though.

Reply:
If the code is put in anyway as suggested in this PR, it might as well be a public API.

Reply:
I don't think that necessarily follows, personally.

Reply:
Not necessarily, true. But if it's there - why not?

Reply:
Because there's no point in adding a method as a public API when it's not needed. It means that if later we find some other way to implement this we might need to remove XML.escape and introduce a breaking change. Do you need XML.escape? Does someone else requested it? No. So please let's stick to this PRs goal, without adding more and more APIs. We can't keep adding API, we need stabilization for once.